gateway/gateway.go

```diff
@@ -134,7 +135,8 @@ func (conf *Config) Validate(path string) ([]string, error) {
 	if conf.ResetPin == nil {
 		return nil, resource.NewConfigValidationFieldRequiredError(path, "reset_pin")
 	}
-	if conf.Bus != 0 && conf.Bus != 1 {
+
+	if conf.Bus != nil && *conf.Bus != 0 && *conf.Bus != 1 {
```

Comment:
Can you also add a check in Validate to not let people configure both USB and SPI together?

I don't know if the 16-channel gateway can mix and match, but I think the normal hats won't be able to.